apps/dashboard/src/api/activity.ts

```diff
@@ -0,0 +1,51 @@
+import { IActivity, IEnvironment } from '@novu/shared';
+import { get } from './api.client';
+
+export interface IActivityFilters {
+  channels?: string[];
+  templates?: string[];
+  email?: string;
+  subscriberId?: string;
+  transactionId?: string;
+}
+
+interface ActivityResponse {
+  data: IActivity[];
+  hasMore: boolean;
+  pageSize: number;
+}
+
+export function getActivityList(
+  environment: IEnvironment,
+  page = 0,
+  filters?: IActivityFilters
+): Promise<ActivityResponse> {
+  const searchParams = new URLSearchParams();
+  searchParams.append('page', page.toString());
+
+  if (filters?.channels?.length) {
+    searchParams.append('channels', filters.channels.join(','));
+  }
+  if (filters?.templates?.length) {
```

Comment:
Since the template concept doesn't exist in V2 and it's just an internal modelling issue, let's rename this to `workflows` for now as it's customer-facing.

Reply:
```suggestion
  if (filters?.workflows?.length) {
```